### PR TITLE
Note on editing with reply markups

### PR DESCRIPTION
**Pull request by ExalFabu**
Hi, i would like to help and contribute on #3111 
Following the [official documentation](https://core.telegram.org/bots/api#updating-messages), the methods i updated are: 
- `Bot.edit_message_text`
- `Bot.edit_message_caption`
- `Bot.edit_message_media`

I am not adding the note on the method `Bot.edit_message_reply_markup` since I don't think it makes sense in the first place, I hope I'm not wrong.

Screenshots:
edit_message_text

edit_message_caption

edit_message_media

### CODE REVIEW

File: telegram/_bot.py [hunk 1]
```diff
@@ -3374,6 +3374,10 @@ async def edit_message_text(
         """
         Use this method to edit text and game messages.
 
+        Note:
+            It is currently only possible to edit messages without
+            :paramref:`reply_markup` or with inline keyboards
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:3379]:
```suggestion
            :attr:`telegram.Message.reply_markup` or with inline keyboards.
```

Same in the other places

File: telegram/_bot.py [hunk 2]
```diff
@@ -3466,6 +3470,10 @@ async def edit_message_caption(
         """
         Use this method to edit captions of messages.
 
+        Note:
+            Please note, that it is currently only possible to edit messages without
```
Comment (Bibo-Joshi) [inline telegram/_bot.py:3474]:
```suggestion
            It is currently only possible to edit messages without
```

to keep it unified :)

**Review by Bibo-Joshi — CHANGES_REQUESTED** [review]
Hi, thanks for the PR! I left two comments below.

> I am not adding the note on the method `Bot.edit_message_reply_markup` since I don't think it makes sense in the first place, I hope I'm not wrong.

Why do yout think that it doesn't make sense to have the note there? people can still try to edit messages without reply markup of with `ReplyKeyboardMarkup` …

**Comment by ExalFabu** [comment]
> Why do yout think that it doesn't make sense to have the note there? people can still try to edit messages without reply markup of with `ReplyKeyboardMarkup` …

As i understood it, the meaning of the note specifies that you can only edit the text of a message if it does not contain a reply markup, since that method is supposed to be used to edit the reply markup I thought it was kind of unnecessary. 
Anyway I added it since I don't disagree with the "better safe than sorry" philosophy

**Comment by Bibo-Joshi** [comment]
Thank you for the contribution! :)

**Pull request merged**